controllers/registries/replication_controller.go

```diff
@@ -368,15 +351,15 @@ func enumReplicationFilterType(filterType string) (string, error) {
 
 // buildReplicationFromCR returns an API conformed ReplicationPolicy object
 func (r *ReplicationReconciler) buildReplicationFromCR(originalReplication *v1alpha2.Replication) (
-	*legacymodel.ReplicationPolicy, error) {
-	newRep := &legacymodel.ReplicationPolicy{
-		ID:            originalReplication.Status.ID,
-		Name:          originalReplication.Spec.Name,
-		Description:   originalReplication.Spec.Description,
-		DestNamespace: originalReplication.Spec.DestNamespace,
-		Override:      originalReplication.Spec.Override,
-		Enabled:       originalReplication.Spec.Enabled,
-		Deletion:      originalReplication.Spec.ReplicateDeletion,
+	*model.ReplicationPolicy, error) {
+	newRep := &model.ReplicationPolicy{
+		Description:       originalReplication.Spec.Description,
+		DestNamespace:     originalReplication.Spec.DestNamespace,
+		Enabled:           originalReplication.Spec.Enabled,
+		ID:                originalReplication.Status.ID,
```

Comment:
Can it happen that `originalReplication.Status.ID` is (still) nil at this point? And would that be bad?

Reply:
Good catch, this shouldn't even be set. The `ID` - when sent in a creation request to the harbor API - will be ignored any way, as it is effectively "read-only" and will be set by harbor internally.